src/main/kotlin/app/revanced/patches/youtube/misc/customplaybackspeed/patch/CustomPlaybackSpeedPatch.kt

```diff
@@ -0,0 +1,103 @@
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.implementation.BytecodeData
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.util.smali.toInstruction
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.annotations.Dependencies
+import app.revanced.patcher.patch.implementation.BytecodePatch
+import app.revanced.patcher.patch.implementation.misc.PatchResult
+import app.revanced.patcher.patch.implementation.misc.PatchResultSuccess
+import app.revanced.patcher.patch.implementation.misc.PatchResultError
+import app.revanced.patches.youtube.misc.customplaybackspeed.annotations.CustomPlaybackSpeedCompatibility
+import app.revanced.patches.youtube.misc.customplaybackspeed.signatures.ArrayGeneratorSignature
+import app.revanced.patches.youtube.misc.customplaybackspeed.signatures.SpeedLimiterSignature
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.builder.BuilderInstruction
+import org.jf.dexlib2.iface.instruction.ReferenceInstruction
+import org.jf.dexlib2.iface.instruction.NarrowLiteralInstruction
+import org.jf.dexlib2.iface.instruction.OneRegisterInstruction
+import org.jf.dexlib2.iface.reference.MethodReference
+import org.jf.dexlib2.iface.reference.FieldReference
+
+@Patch
+@Name("custom-playback-speed")
+@Description("Allows to change the default playback speed options")
+@Dependencies(dependencies = [IntegrationsPatch::class])
+@CustomPlaybackSpeedCompatibility
+@Version("0.0.1")
+class CustomPlaybackSpeedPatch : BytecodePatch(listOf(
+    ArrayGeneratorSignature, SpeedLimiterSignature
+    )) {
+
+    override fun execute(data: BytecodeData): PatchResult {
+        val arrayGenMethod = ArrayGeneratorSignature.result?.method!!
+        val arrayGenMethodImpl = arrayGenMethod.implementation!!
+
+        val sizeCallIndex = arrayGenMethodImpl.instructions
+            .indexOfFirst { ((it as? ReferenceInstruction)?.reference as? MethodReference)?.name == "size" }
+
+        if(sizeCallIndex == -1) return PatchResultError("Couldn't find call to size()")
+
+        val sizeCallResultRegister = (arrayGenMethodImpl.instructions[sizeCallIndex + 1] as OneRegisterInstruction).registerA
+
+        arrayGenMethodImpl.replaceInstruction(sizeCallIndex + 1, 
+            "const/4 v$sizeCallResultRegister, 0x0".toInstruction()
+        )    
+
+        val (arrayLengthConstIndex, arrayLengthConst) = arrayGenMethodImpl.instructions.withIndex()
+            .first {(it.value as? NarrowLiteralInstruction)?.narrowLiteral == 7 }
+
+        val arrayLengthConstDestination = (arrayLengthConst as OneRegisterInstruction).registerA
+
+        val videoSpeedsArrayType = "Lapp/revanced/integrations/settings/Settings;->videoSpeeds:[F"
```

Comment:
You should use already existing package app.revanced.integrations.videoplayer.videosettings.VideSpeed class for this

Reply:
the was already existing too

Reply:
Yes, but it makes no sense at that place and I want to remove that for cleanup :) Makes no sense to have the stuff at multiple places

Reply:
yeah I'll change it in a moment